fms/modules/embedding.py

```diff
@@ -255,6 +255,8 @@ def load_weights(
     def forward(self, inp, reverse=False):
         # If reverse is False, compute input embeddings. If reverse is True, compute output logits.
         # vocab_idx: b n d if reverse, else b n
+        if self.tie_weights and reverse:
+            inp = inp.chunk(self.world_size, dim=2)[self.rank]
```

Comment:
@ani300 TODO: debug this issue

Reply:
solved by https://github.com/foundation-model-stack/foundation-model-stack/pull/331